File: tiny-event-sourcing-spring-app/src/main/kotlin/ru/quipy/shop/delivery/events/DeliveryCreateEvent.java
package ru.quipy.shop.delivery.events;


import lombok.Getter;
import ru.quipy.core.annotations.DomainEvent;
import ru.quipy.domain.Event;
import ru.quipy.shop.delivery.DeliveryAggregate;
import ru.quipy.shop.delivery.entity.DeliveryEntity;

import java.util.Date;
import java.util.UUID;

@Getter
@DomainEvent(name = "DELIVERY_CREATE_EVENT")
public class DeliveryCreateEvent extends Event<DeliveryAggregate> {
    private UUID id;
    private String address;
    private Long price;
    private Date date;

    public DeliveryCreateEvent(UUID id, String address, Long price, Date date) {
        super();
        this.id = id;
        this.address = address;
        this.price = price;
        this.date = date;
    }
}

File: tiny-event-sourcing-spring-app/src/main/kotlin/ru/quipy/shop/delivery/events/DeliveryCreateEvent.kt
package ru.quipy.shop.delivery.events


import lombok.Getter
import ru.quipy.core.annotations.DomainEvent
import ru.quipy.domain.Event
import ru.quipy.shop.delivery.DeliveryAggregate
import ru.quipy.shop.delivery.entity.DeliveryEntity

import java.util.Date
import java.util.UUID

@Getter
@DomainEvent(name = "DELIVERY_CREATE_EVENT")
class DeliveryCreateEvent(
    val deliveryId: UUID,
    var address: String,
    var price: Long,
    var date: Date,
    var orderId: UUID
) : Event<DeliveryAggregate>(
    name = "DELIVERY_CREATE_EVENT"
)